server/game/cards/Time/Generosity.js

```diff
@@ -0,0 +1,84 @@
+const { CardType, BattlefieldTypes } = require('../../../constants.js');
+const Card = require('../../Card.js');
+
+class Generosity extends Card {
+    setupCardAbilities(ability) {
+        this.play({
+            title: 'Generosity',
+            effect: 'choose from 3 actions',
+            gameAction: ability.actions.chooseAction((context) => ({
+                target: context.player,
+                choices: this.getChoices(ability, context.player, context)
+            })),
+            then: {
+                alwaysTriggers: true,
+                gameAction: ability.actions.chooseAction((context) => ({
+                    target: context.player.opponent,
+                    player: context.player.opponent,
+                    choices: this.getChoices(ability, context.player.opponent, context)
+                })),
+                then: {
+                    alwaysTriggers: true,
+                    gameAction: ability.actions.chooseAction((context) => ({
+                        player: context.player,
+                        target: context.player,
+                        choices: this.getChoices(ability, context.player, context)
+                    }))
+                }
+            }
+        });
+    }
+
+    getChoices(ability, player, context) {
+        const controller = context.player === player ? 'self' : 'opponent';
+        const choices = {
+            Unexhaust: ability.actions.removeExhaustion({
+                amount: 1,
+                promptForSelect: {
+                    activePromptTitle:
+                        'Remove 1 exhaustion token from a Phoenixborn or ready spell',
+                    cardType: [CardType.ReadySpell, CardType.Phoenixborn],
+                    controller,
+                    player,
+                    message: '{0} removes 1 exhaustion token from {1}',
+                    messageArgs: (card) => [player, card]
+                }
+            }),
+            Heal: ability.actions.removeDamage({
+                amount: 2,
+                promptForSelect: {
+                    activePromptTitle: 'Remove 2 damage from a unit or Phoenixborn',
+                    cardType: [...BattlefieldTypes, CardType.Phoenixborn],
+                    controller,
+                    player,
+                    message: '{0} heals 2 damage from {1}',
+                    messageArgs: (card) => [player, card]
+                }
+            }),
+            Draw: ability.actions.draw({ amount: 3, showMessage: true })
+        };
+
+        if (context && context.preThenEvent) {
+            this.modifyChoices(choices, context.preThenEvent);
+            if (context.preThenEvent.context && context.preThenEvent.context.preThenEvent) {
+                this.modifyChoices(choices, context.preThenEvent.context.preThenEvent);
+            }
+        }
+
+        return choices;
+    }
+
+    modifyChoices(choices, event) {
+        if (event.amount === 1) {
```

Comment:
kinda hacky to go by the amount, but it was annoying to dissect the event (especially when the heal and unexhaust events were the same type)